tests/utilTest.cpp

```diff
@@ -0,0 +1,421 @@
+    std::vector<unsigned char> big_bytes(9);
+    libzerocash::convertIntToBytesVector(val, big_bytes);
+    BOOST_CHECK( expected == big_bytes);
+}
+
+BOOST_AUTO_TEST_CASE( testConvertBytesVectorToInt ) {
+    uint64_t val;
+    uint64_t expected;
+    std::vector<unsigned char> bytes;
+
+    bytes = { 0, 0, 0, 0, 0, 0, 0, 0 };
+    expected = 0ULL;
+    val = libzerocash::convertBytesVectorToInt(bytes);
+    BOOST_CHECK( expected == val );
+
+    bytes = { 0, 0, 0, 0, 0, 0, 0, 1 };
+    expected = 1ULL;
+    val = libzerocash::convertBytesVectorToInt(bytes);
+    BOOST_CHECK( expected == val );
+
+    bytes = { 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff };
+    expected = 0xffffffffffffffffULL;
+    val = libzerocash::convertBytesVectorToInt(bytes);
+    BOOST_CHECK( expected == val );
+
+    bytes = { 0x80, 0x00, 0x00, 0x00, 0x80, 0x00, 0x00, 0x01 };
+    expected = 0x8000000080000001ULL;
+    val = libzerocash::convertBytesVectorToInt(bytes);
+    BOOST_CHECK( expected == val );
+
+    bytes = { 0xde, 0xad, 0xbe, 0xef }; // opposite of truncation
+    expected = 0xdeadbeefULL;
+    val = libzerocash::convertBytesVectorToInt(bytes);
+    BOOST_CHECK( expected == val );
+}
+
+BOOST_AUTO_TEST_CASE( testConcatenateTwoBoolVectors ) {
+    std::vector<bool> front = { 0, 1, 0 };
+    std::vector<bool> back = { 1, 0, 1 };
+    std::vector<bool> expected = { 0, 1, 0, 1, 0, 1 };
+    std::vector<bool> actual;
+    libzerocash::concatenateVectors(front, back, actual);
+    BOOST_CHECK( expected == actual );
+}
+
+BOOST_AUTO_TEST_CASE( testConcatenateTwoByteVectors ) {
+    std::vector<unsigned char> front = { 0, 1, 0 };
+    std::vector<unsigned char> back = { 1, 0, 1 };
+    std::vector<unsigned char> expected = { 0, 1, 0, 1, 0, 1 };
+    std::vector<unsigned char> actual;
+    libzerocash::concatenateVectors(front, back, actual);
+    BOOST_CHECK( expected == actual );
+}
+
+BOOST_AUTO_TEST_CASE( testConcatenateThreeBoolVectors ) {
+    std::vector<bool> front = { 0, 1, 0 };
+    std::vector<bool> middle { 1, 1, 1 };
+    std::vector<bool> back = { 1, 0, 1 };
+    std::vector<bool> expected = { 0, 1, 0, 1, 1, 1, 1, 0, 1 };
+    std::vector<bool> actual;
+    libzerocash::concatenateVectors(front, middle, back, actual);
+    BOOST_CHECK( expected == actual );
+}
+
+BOOST_AUTO_TEST_CASE( testConcatenateThreeByteVectors ) {
+    std::vector<unsigned char> front = { 0, 1, 0 };
+    std::vector<unsigned char> middle { 1, 1, 1 };
+    std::vector<unsigned char> back = { 1, 0, 1 };
+    std::vector<unsigned char> expected = { 0, 1, 0, 1, 1, 1, 1, 0, 1 };
+    std::vector<unsigned char> actual;
+    libzerocash::concatenateVectors(front, middle, back, actual);
+    BOOST_CHECK( expected == actual );
+}
+
+BOOST_AUTO_TEST_CASE( testSHA256TestVectors ) {
+    unsigned char actual_hash[32];
+    libzerocash::sha256(sha256_preimage, actual_hash, 3);
+    BOOST_CHECK( memcmp(sha256_hash, actual_hash, 32) );
+}
```

Comment:
If these are failing and the test vectors were generated with full `sha256`, the reason may be because ``libzerocash::sha256`` is *only* the compression function I beileve.

If I'm correct, then let's rename it to ``libzerocash::sha256_compress`` to make this clear. As discussed in a mtg, let's fix ``sha256`` test failures in a followup ticket.